File: src/main/java/com/crm/service/UserService.java
package com.crm.service;

import com.crm.dto.request.RegisterRequest;
import com.crm.dto.request.UserCreateRequest;
import com.crm.dto.response.UserResponse;
import com.crm.util.PagedResponse;
import org.springframework.data.domain.Pageable;

public interface UserService {
    UserResponse create(UserCreateRequest request);

    UserResponse register(RegisterRequest request);

    UserResponse getById(Long id);

    PagedResponse<UserResponse> list(Pageable pageable);

    void delete(Long id);
}

File: crm-frontend/src/services/userService.js
import httpClient from '../api/httpClient'

export const userService = {
  async getAll(page = 0, size = 10) {
    const response = await httpClient.get('/users', {
      params: { page, size },
    })
    return response.data
  },

  async getById(id) {
    const response = await httpClient.get(`/users/${id}`)
    return response.data
  },

  async create(userData) {
    const response = await httpClient.post('/users', userData)
    return response.data
  },

  async delete(id) {
    const response = await httpClient.delete(`/users/${id}`)
    return response.data
  },
}
